docs/advanced.md

```diff
@@ -162,3 +162,61 @@ client = httpx.Client(proxies=proxy)
 # This request will be tunneled instead of forwarded.
 client.get("http://example.com")
 ```
+
+
+## Timeout fine-tuning
+HTTPX library offers many request timeout management options. HTTPX distinguish three types of timeouts: **connect** timeouts, 
+**write** timeouts and **read** timeouts.
+
+* Connect timeout specify time period within which a connection between a client and a server must be established.
+* Write timeout specify the maximum duration before timing out writes of the response.
+* Read timeout specify the maximum duration for reading the entire request.
+
+HTTPX allows you to manage timeout policy freely. You can set timeout parameter at client creation phase or pass it on every request.  
+In a nutshell, user can manage timeouts on three different levels of abstraction:
+
+**Request level:**
+```python
+httpx.get('http://example.com/api/v1/example', timeout=5)
+```
+
+**Client level:**
+```python
+client = httpx.Client(timeout=5)
+client.get('http://example.com/api/v1/example')
+```
+
+**TimeoutConfig level:**  
+Unlike request and client level configuration, TimeoutConfig object gives the ability to configure
+every distinct timeout type separately.
+```python
+timeout = httpx.TimeoutConfig(
+    connect_timeout=5, 
+    read_timeout=5, 
+    write_timeout=5
+)
+
+resp = httpx.get('http://example.com/api/v1/example', timeout=timeout)
+```
+
+By default all types of timeouts are set to 5 second, in order to **disable** them you have to pass ``None`` as timeout parameter.
```

Comment:
Should we split this information into sub-sections as well for clarity?

```markdown
### Default timeouts

By default, all types of timeouts are set to **5 seconds**.

### Disabling timeouts

To disable timeouts, you can pass `None` as a timeout parameter. Note that currently this is not supported by the top-level API.

...
```

Reply:
I also was thinking about it. Fixed